### ISSUE TITLE
MambaBDA-Tiny on the [xBD]

### ISSUE THREAD

**Issue by dfordequan** [ChenHongruixuan/MambaCD#19]
what happened to MambaBDA-Tiny on the [xBD] weights

**Comment by ChenHongruixuan** [comment]
Hi,

Thank you so much for your interest to our work! 

We found some small issues with the xBD labels. Previously the labels were generated by us from the json file. However, we now find that the organizers of xView2 provide the labels (with [target] suffix)  after the contest is over. This is slightly different from our training and testing labels. Even though the F1 values have barely dropped, we intend to re-train the model again on the new labels.

Best,

**Comment by dfordequan** [comment]
Thank you for your swift response. I downloaded that pretrained model before and I am getting this building localization result (for an image augmented from the xBD dataset) when inferring. This doesn't look like a labeling issue when training. Do you know what the issue could be?
![guatemala-volcano_00000000_1](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/bdbce09a-632e-4851-996c-9914078b9b41)
![guatemala-volcano_00000000_pre_disaster](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/8dc76309-cf55-412f-b020-2dd4a3898211)

**Comment by ChenHongruixuan** [comment]
Hi,

Thank you for your question and your interest! This is weird. Firstly, may I ask that are you loading the pre-trained model sucessfully? Secondly, why is the value between -4 and 0? What output are you visualising?

Best,

**Comment by dfordequan** [comment]
<img width="957" alt="image" src="https://github.com/ChenHongruixuan/MambaCD/assets/65136673/8a49000d-6edc-41c3-8610-6e41c512c8a6">
<img width="838" alt="image" src="https://github.com/ChenHongruixuan/MambaCD/assets/65136673/417ae3b3-eea4-43f6-a8e8-4a685f91109f">
Hi, I loaded the pre-trained model (kind of) successfully, but I also got IncompatibleKey message as shown in the image.
For your second question, I am visualizing the values of output 1 channel (output_building) of the BDA model, which is the building localization. What I understand is that it represents how confident the model thinks it's a building.

**Comment by ChenHongruixuan** [comment]
Hi,

May I ask which channel you are visualizing? Please note that the output of the loc branch has two channels, please use argmax to get the final building localization result.

Best,

**Comment by dfordequan** [comment]
The image I uploaded was from channel 1. I also have channel 0 and a mask that was generated using argmax. I initially examined the mask and found that it didn't make sense, which led me to start visualizing these two channels.

Also, may I ask if you found anything abnormal in the "IncompatibleKey" message for the pretrained model?
![guatemala-volcano_00000000_0](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/f07190c2-a4b2-447b-bfef-f94214e79602)
![guatemala-volcano_00000000_1 (1)](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/ebe6ec4d-01f7-41aa-b569-c23f3e76446a)
![guatemala-volcano_00000000_mask](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/729b2980-2b9c-4d6b-8fe0-a6fc9a146ffe)
<img width="921" alt="Screenshot 2024-04-28 at 20 49 47" src="https://github.com/ChenHongruixuan/MambaCD/assets/65136673/29cbfd4c-20ef-4f24-ae18-8b623c116caa">

**Comment by ChenHongruixuan** [comment]
That's quite weird... Please intercept the full information printed at the beginning of the loading model for me. The current result has truncation.

Best,

**Comment by dfordequan** [comment]
Hi,

Sorry for the delayed response. The above image basically shows all the printed information defined in the original repo. Afterwards it started infering and printing some customized information i made.
![guatemala-volcano_00000000_pre_disaster](https://github.com/ChenHongruixuan/Mam
![guatemala-volcano_00000000_1](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/a9655255-e17d-4fb4-b5c3-f9a051f2a934)
![guatemala-volcano_00000000_0](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/d301c31b-49e6-4f20-b52f-9b4ad177197a)
baCD/assets/65136673/6669d8f1-76ea-4dff-9bdf-c6b9bb1b4403)
![guatemala-volcano_00000000_mask](https://github.com/ChenHongruixuan/MambaCD/assets/65136673/51164fc2-544f-4c04-8b5a-b6971ca7a091)

However, I've been trying to train the BDA-small model with xBD dataset. After 3000 iterations it already shown some impressive results with the building localization, as shown in the images. Once again, really great work!

For the issue with the pretrained BDA-tiny model. I guess the issues either come from the original vmamba-tiny/config file (as I see they uploaded a new version config in the repo for 'vssm_tiny_0230_ckpt_epoch_262'), or the weights you uploaded. I guess I will wait if you're going to upload a new tiny-BDA weights and try again.

**Comment by ChenHongruixuan** [comment]
Great! Glad to hear that! Thank you again for your interest to our work! 

Best,
